README.md

```diff
@@ -35,6 +45,134 @@ The configuration file is an [HCL](https://github.com/hashicorp/hcl) formatted f
  | `jwt_svid_file_mode`                | The octal file mode to use when saving a JWT SVID file.                                                                           | `0600`                                                                                                                                                               |
+### Operating modes and configuration details
+
+spiffe-helper has two primary operating modes - "daemon mode" (the default),
+where it runs continuously and manages the certificates, and "non-daemon mode",
+where it fetches the certificates once and exits.
+
+The helper can be used in several ways, as detailed below:
+
+* Signal an external process when the certificates are renewed.
+* Run a command that reloads an external process when the certificates are renewed.
+* Run a command that uses the certificates, and signal it when they are renewed
+  (not recommended)
+* Fetch the certificates once and exit.
+
+Note that "daemon mode" does not actually detach itself from the controlling
+process and background itself like a unix "daemon". It just keeps running until
+terminated by a signal or a fatal error.
+
+`spiffe-helper` does not have all the features of a proper process supervisor,
+so it's usually best to let a dedicated process supervisor like systemd manage
+the process that uses the certificates. Use `spiffe-helper` to fetch and renew
+the certificates and wake the external process via `cmd` or `pid_file_name`
+when the certificates are reloaded.
+
+#### Use in daemon-mode with file watcher
+
+If the process using the certificates can watch the on-disk files for changes,
+spiffe-helper can be run in `daemon_mode` with no `cmd` or `pid_file_name`. It
+will fetch and renew the certificates, overwrite them on disk, and the process
+will pick up the changes.
+
+This approach is recommended where possible.
+
+#### Use in daemon-mode with `cmd` to run a process or a reload command
+
+`cmd` and `cmd_args` are used in `daemon_mode` to run a command whenever the
+certificates are renewed. This can be a long-lived process that uses the
+certificates, or a short-lived command that signals a reload mechanism
+for an externally-managed process.
+
+:warning: **cmd_args is not parsed according to shell-like rules**. The
+`cmd_args` will be split into individual arguments using space separation
+unless the argument is enclosed in double quotes, which are consumed. Double
+quotes must be backslash escaped in the hcl string. For example:
+
+```hcl
+cmd_args = "\"this is one argument\""
+```
+
+Double quotes within the argument string must be escaped by doubling them so
+they are not interpreted as argument delimiters. E.g.:
+
+```hcl
+cmd_args = "\"this is a single argument with ONE double-quote \"\" in it\""
```

Comment:
Thanks for updated the explanation.  To have a single double quote it wouldn't be `\\\"`?

Reply:
No - see the test cases. The CSV parser consumes `""` -> `"`, not `\"` -> `"`

Reply:
Thanks for the explanation